### PR TITLE
Fix build against LLVM 14

### PR DESCRIPTION
**Pull request by vchuravy**
Tested against:

```
USE_BINARYBUILDER_LLVM=0
LLVM_GIT_URL:=https://github.com/llvm/llvm-project.git
LLVM_VER=14.0.0
override LLVM_SHA1=4dc3cb8e325525214fccef33e32891e5dccc9540
```

### CODE REVIEW

File: src/Makefile [hunk 1]
```diff
@@ -127,7 +127,7 @@ else
 ifeq ($(OS), Darwin)
 CG_LLVMLINK += $(LLVM_LDFLAGS) -lLLVM
 else
-CG_LLVMLINK += $(LLVM_LDFLAGS) -lLLVM-13jl
+CG_LLVMLINK += $(LLVM_LDFLAGS) -lLLVM-$(LLVM_VER_MAJ)jl
```
Comment (vchuravy) [inline src/Makefile:130]: @staticfloat the one line that concerns you xD

Comment (staticfloat) [reply src/Makefile:130]: I like it!

Comment (vtjnash) [reply src/Makefile:130]: Where did you get this? You are prohibited from this variable here, since it is in `deps`. This file uses `$(LLVM_CONFIG_HOST)`.

**Review by vtjnash — APPROVED** [review]
looks pretty straightforward to me

File: src/Makefile (hunk 1 above)
Comment (vchuravy) [reply src/Makefile:130]:
```
vchuravy@odin ~/b/julia-llvm14> make print-LLVM_VER_MAJ -C src
make: Entering directory '/home/vchuravy/builds/julia-llvm14/src'
LLVM_VER_MAJ=14
make: Leaving directory '/home/vchuravy/builds/julia-llvm14/src'
```

File: src/Makefile [hunk 2]
```diff
@@ -127,7 +127,7 @@ else
 ifeq ($(OS), Darwin)
 CG_LLVMLINK += $(LLVM_LDFLAGS) -lLLVM
 else
-CG_LLVMLINK += $(LLVM_LDFLAGS) -lLLVM-13jl
+CG_LLVMLINK += $(LLVM_LDFLAGS) $(shell $(LLVM_CONFIG_HOST) --libs  --link-shared)
```
Comment (vchuravy) [inline src/Makefile:130]: @vtjnash any clue why we didn't do this before?

Comment (vtjnash) [reply src/Makefile:130]: It used to return the wrong answer

Comment (vtjnash) [reply src/Makefile:130]: Or maybe it still does?
```
llvm-config: error: libLLVM-13jl.dll is missing
```

Comment (vchuravy) [reply src/Makefile:130]: Isn't this the issue that on Windows `dll` needs to be next to the library and `llvm-config` lives in `/tools` and not `/bin`?

**Pull request merged**